src/unitxt/standard.py

```diff
@@ -50,12 +52,35 @@ def verify(self):
                 raise ValueError(
                     f"demos_pool_size must be bigger than num_demos={self.num_demos}, Got demos_pool_size={self.demos_pool_size}"
                 )
+            if self.loader_limit and self.demos_pool_size > self.loader_limit:
+                raise ValueError(
+                    f"demos_pool_size must be bigger than loader_limit{self.loader_limit}, Got demos_pool_size={self.demos_pool_size}"
+                )
```

Comment:
this one is not in the general if

Reply:
Right, because it also requires multi  conditions on the poll size number of demos, which are in the inclosing if statements.